src/libutil/experimental-features.cc

```diff
@@ -5,29 +5,131 @@
-    { Xp::DiscardReferences, "discard-references" },
+struct ExperimentalFeatureDetails
+{
+    ExperimentalFeature tag;
+    std::string_view name;
+    std::string_view description;
 };
 
+constexpr std::array<ExperimentalFeatureDetails, 11> xpFeatureDetails = {{
+    {
+        .tag = Xp::CaDerivations,
+        .name = "ca-derivations",
+        .description = R"(
+            Allows derivations to be content-addressed in order to prevent rebuilds
+            when changes to the derivation do not result in changes to the
+            derivation's output. See
+            [__contentAddressed](../language/advanced-attributes.md#adv-attr-__contentAddressed)
+            for more info.
+        )",
+    },
+    {
+        .tag = Xp::ImpureDerivations,
+        .name = "impure-derivations",
+        .description = R"(
+            Allows derivations to produce non-fixed outputs by setting the `__impure`
+            derivation attribute to `true`. See [these release
+            notes](../release-notes/rl-2.8.md) for an example.
+        )",
+    },
+    {
+        .tag = Xp::Flakes,
+        .name = "flakes",
+        .description = R"(
+            Allows for derivations to be packaged in flakes. See the manual entry for
+            [`nix flake`](../command-ref/new-cli/nix3-flake.md) or this [detailed
+            introduction](https://www.tweag.io/blog/2020-05-25-flakes/) for more info.
+        )",
+    },
+    {
+        .tag = Xp::NixCommand,
+        .name = "nix-command",
+        .description = R"(
+            Allows the usage of the new `nix` CLI subcommands, such as `nix build`, `nix
+            develop`, `nix run`, etc. See the manual for
+            [`nix`](../command-ref/new-cli/nix.md) for more info.
+        )",
+    },
+    {
+        .tag = Xp::RecursiveNix,
+        .name = "recursive-nix",
+        .description = R"(
+            Allow Nix derivations to call Nix in order to recursively build derivations.
+            See [this
+            commit](https://github.com/edolstra/nix/commit/1a27aa7d64ffe6fc36cfca4d82bdf51c4d8cf717)
+            for more info.
+        )",
+    },
+    {
+        .tag = Xp::NoUrlLiterals,
+        .name = "no-url-literals",
+        .description = R"(
+            Disallows unquoted URLs as part of the Nix language syntax. See [RFC
+            45](https://github.com/NixOS/rfcs/pull/45) for more info.
+        )",
+    },
+    {
+        .tag = Xp::FetchClosure,
+        .name = "fetch-closure",
+        .description = R"(
+            Enables the use of the `fetchClosure` function in the standard library. See
+            the docs for [`fetchClosure`](../language/builtins.md#builtins-fetchClosure)
+            for more info.
+        )",
+    },
+    {
+        .tag = Xp::ReplFlake,
+        .name = "repl-flake",
+        .description = R"(
+            Allows the user to enter a Nix REPL within a flake, e.g. `nix repl nixpkgs`
+            or `nix repl .#foo`.
```

Comment:
```suggestion
            Allow passing [installables](@docroot@/command-ref/new-cli/nix.md#installables) to `nix repl`, making its interface consistent with the other experimental commands.
```

@nixos/nix-team that experimental feature flag could as well be folded into `nix-command` by changing [one line](https://github.com/NixOS/nix/blob/d93c1514ebcf71390c8a8070759487e5a40dcc6f/src/nix/repl.cc#L49). What do you think? This PR is great because it trivially exposes such potential for simplification.